From 06e2ecd0df9b81dd518ae6017ec42765520e2e93 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Thu, 20 Sep 2012 21:49:40 -0700
Subject: [PATCH 1/3] XQuartz: Avoid a possible deadlock with DRI on OS X
 10.7.5 and OS X 10.8.2

<rdar://problem/12338921>
http://bugs.winehq.org/show_bug.cgi?id=31751

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 25d26875bc9bd6fd23ae1b5280f015abf1b033b7)
---
 hw/xquartz/xpr/dri.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git xorg-server-1.13.1/hw/xquartz/xpr/dri.c xorg-server-1.13.1/hw/xquartz/xpr/dri.c
index 03af163..adba69c 100644
--- xorg-server-1.13.1/hw/xquartz/xpr/dri.c
+++ xorg-server-1.13.1/hw/xquartz/xpr/dri.c
@@ -64,6 +64,7 @@
 #include "mi.h"
 #include "mipointer.h"
 #include "rootless.h"
+#include "rootlessCommon.h"
 #include "x-hash.h"
 #include "x-hook.h"
 #include "driWrap.h"
@@ -380,6 +381,11 @@ DRICreateSurface(ScreenPtr pScreen, Drawable id,
     DRIDrawablePrivPtr pDRIDrawablePriv;
 
     if (pDrawable->type == DRAWABLE_WINDOW) {
+        /* <rdar://problem/12338921>
+         * http://bugs.winehq.org/show_bug.cgi?id=31751
+         */
+        RootlessStopDrawing((WindowPtr)pDrawable, FALSE);
+
         pDRIDrawablePriv = CreateSurfaceForWindow(pScreen,
                                                   (WindowPtr)pDrawable, &wid);
 
-- 
1.8.0.2

